M2/Macaulay2/packages/WhitStrat.m2

```diff
@@ -0,0 +1,268 @@
+newPackage(
+	"WhitStrat",
+	Version => "1.1", 
+    	Date => "May 24, 2022",
+    	Authors => {{Name => "Martin Helmer", 
+		  Email => "mhelmer@ncsu.edu", 
+		  HomePage => "http://martin-helmer.com/"}},
+    	Headline => "Compute Whitney Statifications Algebraically",
+    	DebuggingMode => false,
+	Reload => true
+    	);
+needsPackage "Elimination";
+needsPackage "PrimaryDecomposition";
+needsPackage "Saturation";
+export{
+    "conormal",
+    "conormRing",
+    "whitneyStratify"
+    }
+
+conormRing=method(TypicalValue=>Ring);
+conormRing(Ideal):=(I)->(
+     R:=ring I;
+    n:=numgens(R)-1;
+    kk:=coefficientRing R;
+    degs:=join(for i from 0 to n list {1,0},for i from 0 to n list {0,1});
+    v:=symbol v;
+    S:=kk[gens(R),v_0..v_n,Degrees=>degs];
+    return S;
+
+    );
+
+conormRing(Ideal,List,List):=(I,Alist,alpha)->(
+    R:=ring I;
+    n:=numgens(R)-1;
+    kk:=coefficientRing R;
+    degs:=join(for i from 0 to n list {1,0},for i from 0 to n list {0,1});
+    indset:=for vert in alpha list position(Alist,i->i==vert);
+    v:=symbol v;
+    S:=kk[gens(R),for i in indset list v_i,Degrees=>degs];
+    print S;
+    return S;
+
+    );
+
+conormal=method(TypicalValue=>Ideal);
+conormal (Ideal):=(I)->(
+    return conormal(I,conormRing(I));
+    );
+conormal (Ideal,List,List):=(I,A,alpha)->(
+    return conormal(I,conormRing(I,A,alpha));
+    );
+conormal (Ideal,List,List,Ring):=(I,A,alpha,S)->(
+    N:=conormal(I,conormRing(I,A,alpha));
+    S1:=ring(N);
+    indset:=for vert in alpha list position(A,i->i==vert);
+    xN:=select(gens(S1),i->degree(i)=={1,0});
+    vN:=select(gens(S1),i->degree(i)=={0,1});
+    xS:=select(gens(S),i->degree(i)=={1,0});
+    vS:=select(gens(S),i->degree(i)=={0,1});
+    subList:={};
+    count:=0;
+    for i in indset do (
```

Comment:
Instead of appending one element at a time to this initially empty list, which turns an order N algorithm into an order N^2 algorithm (because "append" copies the whole list) why not use "for ... list .... " instead?